src/projects/TodoList/index.js

```diff
@@ -13,23 +15,33 @@ class TodoList extends React.Component {
   constructor() {
     super();
     this.addItem = this.addItem.bind(this)
+    this.removeItem = this.removeItem.bind(this)
     this.state = {
       todos: ['Take out the bins', 'Put a wash load on', 'Learn React'],
-      newItem: ''
+      newItem: '',
+      removedIndex: ''
     }
   }
   addItem() {
     const newTodos = this.state.todos
     newTodos.push(this.state.newItem)
     this.setState({ todos: newTodos, newItem: '' })
   }
+  removeItem() {
+    let removedItem = this.props.key
+    this.setState({ removedIndex: removedItem })
+    const removedTodos = this.state.todos
+    removedTodos.splice(this.state.todos[this.state.removedIndex], 1)
+    this.setState({ todos: removedTodos, removedIndex: '' })
+  }
   render() {
     // Before the return you can do all logic you need
     let todos = this.state.todos.map((task, i) => {
       return (
         <Todo
           key={i}
           task={task}
+          removeItem={this.removeItem}
```

Comment:
For each iteration of the map, `i` will equal a different number  relating to the current index of the item it's mapping over. What we're doing here is storing a function in memory to be executed when the button is clicked but that function is stored with a different `i` value for each `<Todo />` that is returned. 

```js
<Todo
  key={i}
  task={task}
  removeItem={() => {
    this.removeItem(i);
  }}
/>
```

If you imagine an array of 3 todos which has been mapped, will look something like this:

```js
return (
    <Todo key={0} task="Take out bins" removeItem={() => { this.removeItem(0) }} />
    <Todo key={1} task="Do washing" removeItem={() => { this.removeItem(1) }} />
    <Todo key={2} task="Git gud" removeItem={() => { this.removeItem(2) }} />
)
```

This means if you now pass an argument `i` into your `removeItem` function, when its called by the click of any one of those Todos it will reference that Todos index in the todos array (allowing you to remove it safely).